addons/sale_product_configurator/static/src/js/product_configurator_controller.js

```diff
@@ -52,6 +52,12 @@ var ProductConfiguratorFormController = FormController.extend({
         if (this.renderer.state.context.configuratorMode === 'options') {
             this.$el.closest('.modal').addClass('d-none');
         }
+
+        if (this.renderer.displayOptModal && this.renderer.state.context.configuratorMode !== 'edit') {
+            this.$el.closest('.o_dialog_container').addClass('d-none');
+        } else {
+            this.$el.closest('.o_dialog_container').removeClass('d-none');
+        }
```

Comment:
```suggestion
        this.$el.closest('.o_dialog_container').toggleClass(
            'd-none',
            this.renderer.displayOptModal && this.renderer.state.context.configuratorMode !== 'edit')
        );
```
or
```javascript
const myvar =  this.renderer.displayOptModal && this.renderer.state.context.configuratorMode !== 'edit';
this.$el.closest('.o_dialog_container').toggleClass( 'd-none', myvar);
```

To check, condition might be the otherway around, but avoid duplicating lines when you can :)